src/wms/src/main/java/org/geoserver/wms/legendgraphic/LegendMerger.java

```diff
@@ -0,0 +1,795 @@
+                    LegendUtils.getLabelFont(req), forceLabelsOn, forceLabelsOff);
+        }
+    }
+    
+    /**
+     * Receives a list of <code>BufferedImages</code> and produces a new one which holds all the images in <code>imageStack</code> one above the
+     * other.
+     * 
+     * @param imageStack the list of BufferedImages
+     * @param dx horizontal space between images
+     * @param dy vertical space between images
+     * @param margin padding around image
+     * @param backgroundColor background color of legend
+     * @param transparent if true make legend transparent
+     * @param antialias if true applies anti aliasing
+     * @param layout orientation of legend, my be horizontal or vertical
+     * @param rowWidth maximum width for horizontal legend
+     * @param rows maximum number of rows for horizontal legend
+     * @param columnHeight maximum height for vertical legend
+     * @param columns maximum number of columns for vertical legend
+     * @return the legend image with all the images on the argument list.
+     */
+    public static BufferedImage mergeRasterLegends(MergeOptions mergeOptions) {
+        List<RenderedImage> imageStack= mergeOptions.getImageStack();
+        LegendLayout layout = mergeOptions.getLayout();
+        
+        List<BufferedImage> nodes = new ArrayList<BufferedImage>();
+        final int imgCount = imageStack.size();
+        for (int i = 0; i < imgCount; i++) {
+            nodes.add((BufferedImage) imageStack.get(i));
+        }
+
+        BufferedImage finalLegend = null;
+        if (layout == LegendLayout.HORIZONTAL) {
+            Row[] rows = createRows(nodes, mergeOptions.getRowWidth(), mergeOptions.getRows());
+            finalLegend = buildFinalHLegend(mergeOptions, rows);
+        }
+
+        if (layout == LegendLayout.VERTICAL) {
+            Column[] columns = createColumns(nodes, mergeOptions.getColumnHeight(), mergeOptions.getColumns());
+            finalLegend = buildFinalVLegend(mergeOptions, columns);
+        }
+
+        return finalLegend;
+    }
+
+    /**
+     * Receives a list of <code>BufferedImages</code> and produces a new one which holds all the images in <code>imageStack</code> one above the
+     * other, handling labels.
+     * 
+     * @param imageStack the list of BufferedImages, one for each applicable Rule
+     * @param rules The applicable rules, one for each image in the stack (if not null it's used to compute labels)
+     * @param request The request.
+     * @param forceLabelsOn true for force labels on also with a single image.
+     * @param forceLabelsOff true for force labels off also with more than one rule.
+     * 
+     * @return the image with all the images on the argument list.
+     * 
+     */
+    public static BufferedImage mergeLegends(MergeOptions mergeOptions, Rule[] rules, GetLegendGraphicRequest req) {

```

Comment:
I would always list mergeOptions as the last parameter in the list, since it is open ended ...